### ISSUE TITLE
Review HTML sanitizer.

### ISSUE THREAD

**Issue by afshin** [jupyterlab/jupyterlab#4091]
[In this thread](https://github.com/cython/cython/issues/1775#issuecomment-369918244), it is demonstrated that an `onClick` attribute on a button can come through. This seems like a bug.

```

from IPython.core.magic import Magics, magics_class, line_magic
from IPython.display import display, HTML
toggleDiv = """
this.nextElementSibling.style.display != 'block' ? 
this.nextElementSibling.style.display = 'block' : 
this.nextElementSibling.style.display = 'none'
"""

HTML(f"""
    <button onClick="{toggleDiv}">Click</button>
    <div style='display=none'>ok</div>""")
```

**Comment by ian-r-rose** [comment]
It looks to me like it is getting through because the outputs are considered [trusted](https://github.com/jupyterlab/jupyterlab/blob/master/packages/rendermime/src/renderers.ts#L56).

What is the intended behavior here? This seems like a pretty big escape hatch for inline javascript.

**Comment by blink1073** [comment]
Hmm, we had been relying on the default behavior of browsers not to execute `<script>` tags when `html` is added.  I think this would require a "strip script" form of santization in addition to our full sanitization.

**Comment by jasongrout** [comment]
I think we discussed this at one point (@sccolbert, @ellisonbg at least, at PyData Seattle). We realized that we were not completely preventing js for trusted content, but we were making it harder to include js, raising the bar, to discourage it.

Because the content is trusted, we aren't so concerned about security. I think here it is a matter of *encouraging* people to write proper plugins for the javascript execution.

**Comment by akhmerov** [comment]
With RTC the trust model might not be sufficient. As it stands right now, jupyterlab-google-drive is vulnerable to this js injection. AFAIK that's something the current design aimed to avoid.

**Comment by jasongrout** [comment]
Can you elaborate? I think the assumption is that whoever you are collaborating with is a trusted collaborator.

**Comment by jasongrout** [comment]
Or maybe more precisely - I think the assumption in the google collaboration plugin is that if an output is trusted for any person, it is trusted for all collaborators.

**Comment by akhmerov** [comment]
Unfortunately I don't remember the exact source. IIRC when I asked about the argument for blocking arbitrary js from being rendered, someone from the dev team used RTC as one of the strong arguments for doing so.

Your description of the assumption of the google-drive plugin is indeed correct. However I am concerned about this as a threat model that is easy to grasp: "trusted collaborator who can execute arbitrary code without your knowledge" might not be an obvious assumption for the users.

**Comment by ian-r-rose** [comment]
I don't think we have really made a decision about how much to trust collaborators, beyond some exploratory conversations. It is certainly a concern, and one can envision different situations where you might have more or less trust.

I'm not sure we should automatically assume they are trusted.

**Comment by jasongrout** [comment]
I'm definitely interested in other sets of assumptions about trust/threat models - just saying what I thought the current assumption was.

**Comment by BoPeng** [comment]
I am presenting SoS and one of the notebooks does not function as expected because all of my icons stopped working. The console window shows 

```
HTML.sanitizer i.onclick removed
```
which leads me to this ticket. What should I do to correct this?

Briefly speaking, the SoS kernel outputs HTML tables with a bunch of clickable icons, which, among other things, sends messages back to the kernel to query the status of workflows. These icons worked well before until this sanitizer thing hits.

![image](https://user-images.githubusercontent.com/9889312/49884430-75837180-fdfa-11e8-98b3-b03f2a4c9d86.png)
